includes/compat/class-wc-stripe-subscriptions-helper.php

```diff
@@ -114,6 +110,39 @@ public static function get_detached_subscriptions( $limit = -1 ) {
 		return $detached_subscriptions;
 	}
 
+	/**
+	 * Checks if a subscription's payment method is detached from the customer.
+	 *
+	 * @param $subscription WC_Subscription The subscription object to check.
+	 * @return bool True if the payment method is detached, false otherwise.
+	 */
+	public static function is_subscription_payment_method_detached( $subscription ) {
+		$source_id = $subscription->get_meta( '_stripe_source_id' );
+		if ( ! $source_id ) {
+			return false;
+		}
+
+		$payment_method = WC_Stripe_API::get_payment_method( $source_id );
+		if ( is_wp_error( $payment_method ) ) {
+			// If we can't retrieve the payment method, assume it's detached.
+			WC_Stripe_Logger::error(
+				sprintf(
+					/* translators: %1$s is the subscription ID, %2$s is the error message */
+					__( 'Error retrieving payment method for subscription %1$s: %2$s', 'woocommerce-gateway-stripe' ),
+					$subscription->get_id(),
+					$payment_method->get_error_message()
```

Comment:
Noting for caution, when the error is not a wp_error object but the error I mentioned in the above comment (`$payment_method->error`), this line will throw a fatal error for undefined `get_error_message` function.